src/services/recipient.js

```diff
@@ -481,30 +504,41 @@ export function reduceObjectivesForRecipientRecord(
     });
 
   const current = goal;
-  current.goalTopics = reduceTopicsOfDifferingType([...goal.goalTopics, ...topics]);
+  const goalTopics = Array.isArray(goal.goalTopics) ? goal.goalTopics : [];
+  const safeTopics = ensureArray(topics);
+  const goalReasons = Array.isArray(goal.reasons) ? goal.reasons : [];
+  const safeReasons = ensureArray(reasons);
+
+  current.goalTopics = reduceTopicsOfDifferingType([...goalTopics, ...safeTopics]);
   current.goalTopics.sort();
 
-  current.reasons = uniq([...goal.reasons, ...reasons]);
+  current.reasons = uniq([...goalReasons, ...safeReasons]);
   current.reasons.sort();
 
   return [...objectives].map((obj) => {
     // eslint-disable-next-line no-param-reassign
     obj.topics = reduceTopicsOfDifferingType(obj.topics);
     return obj;
-  }).sort((a, b) => ((
-    a.endDate === b.endDate ? a.id < b.id
-      : new Date(a.endDate) < new Date(b.endDate)) ? 1 : -1));
+  }).sort((a, b) => {
+    const dateA = a?.endDate?.trim() ? new Date(a.endDate) : new Date('1970-01-01');
+    const dateB = b?.endDate?.trim() ? new Date(b.endDate) : new Date('1970-01-01');
```

Comment:
this will not be good when a.endDate = 'A'

Reply:
Added handling of invalid dates.